### PR TITLE
[POWEROPS-2558] chore: linters and codecov in workflows

### PR DESCRIPTION
**Pull request by nodegard**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/power-ops-sdk/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [pyproject.toml](https://github.com/cognitedata/power-ops-sdk/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -50,26 +55,33 @@ jobs:
 
       - name: Test
         env:
-          CI: 1
-        run: pytest --skipcdf
+          SETTINGS__COGNITE__CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
+          SETTINGS__COGNITE__CLIENT_ID: ${{ vars.CLIENT_ID }}
+          SETTINGS__COGNITE__PROJECT: ${{ vars.PROJECT }}
+          SETTINGS__COGNITE__CDF_CLUSTER: ${{ vars.CLUSTER }}
+          SETTINGS__COGNITE__TENANT_ID: ${{ vars.TENANT_ID }}
```
Comment (eriklien) [inline .github/workflows/build.yml:62]: **question**: Is this pointing to [environment variables in the CI environment](https://github.com/cognitedata/power-ops-sdk/settings/environments/1542063228/edit#:~:text=10%20months%20ago-,Environment%20variables,-Variables%20are%20used)? I don't see them there.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,26 +8,6 @@ env:
...
-      - name: Install required dependencies
-        run: |
-          python3 -m pip install --upgrade pip poetry
-          poetry config virtualenvs.create false
-          poetry install
-
-      - name: Linting and static code checks
-        run: |
-          pre-commit run --all-files
-
```
Comment (eriklien) [inline .github/workflows/release.yml:30]: **question**: Not needed here because we do it in build.yml?

**Review by eriklien — APPROVED** [review]
Looks good. Just added a few questions for my understanding.

File: .github/workflows/build.yml (hunk 1 above)
Comment (nodegard) [reply .github/workflows/build.yml:62]: They're on the repo level because all the envs need them

File: .github/workflows/release.yml (hunk 2 above)
Comment (nodegard) [reply .github/workflows/release.yml:30]: Yea, no need to re-run pre-commit on merge to main given the branch protections and strict requirements for passing on PRs toward main

**Pull request merged**